### PR TITLE
Remove tex version check; require dvipng >=1.6

### LINKED ISSUE #8748: Use of recent Miktex version (2.9.6350) with Matplotlib --> "tex not found"
### Bug report

**Bug summary**

During use of Miktex (v2.9.6350) together with Python (v2.7) and Matplotlib (v2) on Windows 10, Matplotlib checks for the installed Tex version during initialization via the "tex -version" command. The command line call "tex -version" hands back only the Miktex version (in previous Miktex versions it handed back Miktex AND Tex version) which leads to Matplotlib thinking that the installed Tex version is outdated (it simply checks for the number 3.1). 

The result is that when "text.usetex" = True in the matplotlibrc file, Matplotlib issues a warning during initialization as it cannot find an update tex version. Setting "text.usetex" = True with dynamic rc settings works fine as no check for a current tex version is performed.

> C:\ProgramData\Anaconda2\lib\site-packages\matplotlib\__init__.py:513: UserWarning: matplotlibrc text.usetex option can not be used unless TeX-3.1415 or later is installed on your system

### PR DESCRIPTION
**Pull request by anntzer**
Fixes #8748, plus some cleanups.

### CODE REVIEW

File: lib/matplotlib/__init__.py [hunk 1]
```diff
@@ -390,6 +390,9 @@ def checkdep_ghostscript():
 checkdep_ghostscript.version = None
 
 
+# Deprecated, as it is unneeded and some distributions (e.g. MiKTeX 2.9.6350)
+# do not actually report the TeX version.
+@cbook.deprecated("2.1")
```
Comment (QuLogic) [inline lib/matplotlib/__init__.py]: This is used in tests also.

File: pytest.ini [hunk 2]
```diff
@@ -14,9 +14,12 @@ pep8ignore =
     setup.py E402 E501
     setupext.py E301 E302 E501
     setup_external_compile.py E302 E501 E711
+
     versioneer.py ALL  # External file.
     tools/gh_api.py ALL  # External file.
     tools/github_stats.py ALL  # External file.
+    lib/matplotlib/cbook/_backports.py ALL  # Copy-pasted functions.
```
Comment (QuLogic) [inline pytest.ini]: No `lib/`, as you can see below.

**Comment by anntzer** [comment]
both issues addressed

File: lib/matplotlib/animation.py [hunk 3]
```diff
@@ -416,22 +415,10 @@ def bin_path(cls):
     @classmethod
     def isAvailable(cls):
         '''
-        Check to see if a MovieWriter subclass is actually available by
-        running the commandline tool.
+        Check whether the command-line tool for a MovieWriter subclass actually
+        exists in the PATH.
         '''
-        bin_path = cls.bin_path()
```
Comment (tacaswell) [inline lib/matplotlib/animation.py]: This is here to let sub-classes apply some extra filtering, please revert this change.

Comment (anntzer) [inline lib/matplotlib/animation.py]: Ah I had missed that, removed the commit.
